contrib/crc32-vpmsum-cmake/vec_crc32.h

```diff
@@ -0,0 +1,29 @@
+#ifndef VEC_CRC32
+#define VEC_CRC32
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+unsigned int crc32_vpmsum(unsigned int crc, const unsigned char *p, unsigned long len);
+
+static inline uint32_t crc32_ppc(uint64_t crc, unsigned char const *buffer, size_t len)
+{
+	unsigned char *emptybuffer;
+    if (!buffer) {
```

Comment:
Sure that we should allow the caller to pass an empty buffer? Doing an on-the-fly allocation for each CRC calculation sounds quite inefficient (vs. re-using the same buffer). Could you please check if the CRC implementations on other platforms have the same logic?

Reply:
The above code has been added to handle the error handling. In the other CRC implementations we don't see the same logic.

Reply:
Are you saying that the on-the-fly allocation block can go away? If not, could you please give an example where such code is actually needed?

(Asking because for platforms x86 and ARM, intrinsic are called (`_mm_crc32_u64`/`__crc32cd`) which don't have such logic. Perhaps you could add an `assert` that buffer is set.)

Reply:
Yes, we will replace the error handle with assert.